### PR TITLE
Add checks to URL dupe checker

### PR DESCRIPTION
**Pull request by arandomandy**
This PR includes two features:

1. It makes the URL dupe checker ignore URLs with anchors, so that scenario A is covered
2. It makes the URL dupe checker consider scenario B, it does so by appending a # to the URL and searching for a pattern where it matches the URL plus any characters after the #

Scenario A: Alice submits the following URL https://artagnon.com/articles/leancoq, three minutes later Bob tries submitting https://artagnon.com/articles/leancoq#main

Scenario B: Alice submits the following URL https://artagnon.com/articles/leancoq#main, three minutes later Bob tries submitting https://artagnon.com/articles/leancoq

### CODE REVIEW

File: app/models/story.rb [hunk 1]
```diff
@@ -229,10 +230,16 @@ def self.find_similar_by_url(url)
     end
     urls = urls2.uniq
 
+    # trailing pound
+    urls.each do |u|
+      urls_with_trailing_pound.push u + "#"
```
Comment (pushcx) [inline app/models/story.rb]: Is this just for the case where the url already in the database is `https://example.com/foo#`, right? If so, you can remove it - we only have 29 of them. Let's just write a migration to remove them.

File: app/models/story.rb [hunk 2]
```diff
@@ -229,10 +230,16 @@ def self.find_similar_by_url(url)
...
+    end
+
     # if a previous submission was moderated, return it to block it from being
     # submitted again
     Story
       .where(:url => urls)
+      .or(Story.where("url RLIKE ?", urls_with_trailing_pound.join(".|")))
```
Comment (pushcx) [inline app/models/story.rb]: What is the `.` for in this `join`?

File: app/models/story.rb (hunk 1 above)
Comment (arandomandy) [reply app/models/story.rb]: No, this is instead to take care of the scenario described below. It does so by appending a `#` to the URL and searching for a pattern where it matches the URL plus any characters after the `#`, thus the ugly and expensive `RLIKE`.

Alice submits the following URL https://artagnon.com/articles/leancoq#main, three minutes later Bob tries submitting https://artagnon.com/articles/leanco

File: app/models/story.rb (hunk 2 above)
Comment (arandomandy) [reply app/models/story.rb]: Short answer: 

Pattern matching: `.` matches any character.

Long answer:

So, let's say Alice submits the following URL https://artagnon.com/articles/leancoq#main, three minutes later Bob tries submitting https://artagnon.com/articles/leancoq

How do I know if https://artagnon.com/articles/leancoq has been submitted with an anchor before, i.e. https://artagnon.com/articles/leancoq#main? I first [added a `#` to all URLs](https://github.com/lobsters/lobsters/pull/802/files#diff-8a4ad5774f65a8a217b216fa2038a5bcR235), then I try to find any URL that matches the following pattern; https://artagnon.com/articles/leancoq#anycharacter

I'm able to do so by appending a `.` at the end of each URL.

As I pointed out [here](https://github.com/lobsters/lobsters/issues/801#issuecomment-571098708) I have my doubts about this approach.

**Comment by pushcx** [comment]
Thanks for doing this, I'm glad to start catching these at submission.

**Pull request merged**